github/resource_github_branch_default.go

```diff
@@ -63,6 +65,14 @@ func resourceGithubBranchDefaultRead(d *schema.ResourceData, meta interface{}) e
 
 	repository, _, err := client.Repositories.Get(ctx, owner, repoName)
 	if err != nil {
+		if ghErr, ok := err.(*github.ErrorResponse); ok {
+			if ghErr.Response.StatusCode == http.StatusNotFound {
+				log.Printf("[INFO] Removing branch_default for repository %s/%s from state because the repository no longer exists in GitHub",
```

Comment:
hmm my main concern with how the PR is right now is that it proposes a change in behavior on `resourceGithubBranchDefaultRead` which seems very general purpose (reading defaul branches), so eliding the error when the response is status not found may break other things that depend on this, even though it could solve for the archival issue. :thinking: 

Would be better if we can find a way to solve the issue closer to the logic that depends on this specific check (if the error is status not found). Will check the code to see if I can find something, I'm not familiar with the provider code.

Also in general it is better to use [errors.As](https://pkg.go.dev/errors#As) to cast errors since it will allow you to unwrap the error by type instead of a direct cast that won't work if the error is wrapped with additional context.

Reply:
this is just the function terraform is using to refresh the actual state of the resource... so by setting the id to "" we tell terraform that resource vanished and doe snot exist anymore... the error message is just a hint here and won't shown in a plan afaics it is the same with a repository that vanished in the repository resource.

Reply:
hmm yet they could have some tests that ensure that in this situation it errors instead of setting to ""/resource vanished and that test/behavior would break, lets setup the basics to run the tests so we can check the tested behavior and add a new test for this new behavior too (improve chances of a change being accepted upstream).